dot/sync/block_importer.go

```diff
@@ -99,6 +108,39 @@ func (b *blockImporter) importBlock(bd *types.BlockData, origin BlockOrigin) (im
 	return true, nil
 }
 
+func (b *blockImporter) checkAndApplyForcedAuthChanges(blockData types.BlockData) error {
+	header := blockData.Header
+	consensusDigests, err := types.FilterGRANDPAForcedChanges(header.Digest.ToConsensusDigests())
+	if err != nil {
+		return fmt.Errorf("failed while checking GRANDPA digests: %w", err)
+	}
+
+	for i := range consensusDigests {
+		digest := consensusDigests[i]
+		if digest.ConsensusEngineID == types.GrandpaEngineID {
+			data := types.NewGrandpaConsensusDigest()
+			err := scale.Unmarshal(digest.Data, &data)
+			if err != nil {
+				return fmt.Errorf("unmarshaling grandpa consensus digest: %w", err)
+			}
+
+			digestValue, err := data.Value()
+			if err != nil {
+				return fmt.Errorf("getting digest value: %w", err)
+			}
+
+			if _, ok := digestValue.(types.GrandpaForcedChange); ok {
+				err = b.grandpaState.HandleGRANDPADigest(header, data)
+				if err != nil {
+					return fmt.Errorf("handling grandpa digest: %w", err)
+				}
+			}
+		}
+	}
+
+	return b.grandpaState.ApplyForcedChanges(header)
```

Comment:
We should not apply the forced changes here, we should let it to the import flow, here we should only check the sigs w/o changing the state.